components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -127,6 +132,37 @@ public static String getSCIMRoleV2URL(String id) {
         return StringUtils.isNotBlank(id) ? getSCIMRoleV2URL() + SCIMCommonConstants.URL_SEPERATOR + id : null;
     }
 
+    /**
+     * Returns the latest SCIM Role URL based on the configuration.
+     * If the configuration USE_ROLE_V2_EP_IN_REF is set to true, it returns the SCIM Role V2 URL.
+     * Otherwise, it returns the SCIM Role V3 URL.
+     *
+     * @param id The ID of the role.
+     * @return The latest SCIM Role URL with the given ID, or null if the ID is blank.
+     */
+    public static String getDefaultSCIMRoleURL(String id) {
+
+        String roleAPIVersion = IdentityUtil.getProperty(DEFAULT_ROLE_API_VERSION_FOR_REF);
+        String roleURL = getSCIMRoleV2URL();
+        if (StringUtils.isNotBlank(roleAPIVersion)) {
+            if (V1.getVersion().equals(roleAPIVersion)) {
+                roleURL = getSCIMRoleURL();
+            } else if (V2.getVersion().equals(roleAPIVersion)) {
+                roleURL = getSCIMRoleV2URL();
+            } else if (V3.getVersion().equals(roleAPIVersion)) {
+                roleURL = getSCIMRoleV3URL();
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("The configured role API version: " + roleAPIVersion + " is not valid. " +
+                            "Hence using the v2 role API version: " + V3.getVersion());
```

Comment:
Btw, isn't that new version should be v3, and need to configure a value in deployment.toml to preserve backward compatibility

Reply:
The configuration defaults to v3. 

However, if there is a configuration issue, the system falls back to v2 to preserve backward compatibility. 

This ensures that existing customers, who are more likely to adjust these configs, won’t face unexpected breakage if something is mis-configured. 

That's why I added it like that. Any suggestions?